frontend/src/pages/Publish.tsx

```diff
@@ -12,16 +12,34 @@ import { useAI } from '../hooks/blog';
 import GenerateAIBtn from '../components/GenerateAIBtn';
 import PublishTags from '../components/PublishTags';
 import { htmlTagRegex } from '../util/string';
+import useTimerInterval from '../hooks/useTimerInterval';
+import { STORAGE_KEY } from '../util/storage-keys';
 
 const Publish = () => {
+  const userJSON = localStorage.getItem('user') || '{}';
+  const user = JSON.parse(userJSON);
+
   const { generateBlog } = useAI();
-  const [title, setTitle] = useState('');
-  const [content, setContent] = useState('');
+  const [title, setTitle] = useState(() => getDraftHandler('title'));
+  const [content, setContent] = useState(() => getDraftHandler('content'));
   const [blogId, setBlogId] = useState('');
 
+  useTimerInterval(15000, autoSaveHandler);
+
   const writingPadRef = useRef<ReactQuill>(null);
 
-  useEffect(() => {}, [content]);
+  function getDraftHandler(key: string) {
+    const draftJSON = localStorage.getItem(STORAGE_KEY.WRITE_DRAFT(user.id));
+    if (draftJSON) {
+      const draft = JSON.parse(draftJSON);
+      return draft[key];
+    } else return '';
+  }
+
+  function autoSaveHandler() {
+    const draft = { title, content };
+    localStorage.setItem(STORAGE_KEY.WRITE_DRAFT(user.id), JSON.stringify(draft));
+  }
```

Comment:
@KrupaPanchal2527 - We would need to perform autosave when user edits existing article as well. Can we move this inside  a hook  so that it becomes reusable ?

Reply:
@aadeshkulkarni 
Okay, I'll make changes accordingly. Should I add an autosave ability to edit in this branch itself?

Reply:
If it's not too much trouble, yes. Otherwise, I can create another issue for later.